packages/types/api/providers.d.ts

```diff
@@ -0,0 +1,32 @@
+export interface ProviderRecord {
+  id: string,
+  globalConnectionLimit?: number,
+  protocol: string,
+  host: string,
+  port?: number,
+  username?: string,
+  password?: string,
+  encrypted?: boolean,
+  createdAt: number,
+  updatedAt?: number,
+  privateKey?: string,
+  cmKeyId?: string,
+  certificateUri?: string
+}
+
+export interface PostgresProviderRecord {
+  cumulusId?: number,
+  name: string,
+  globalConnectionLimit?: number,
+  protocol: string,
+  host: string,
+  port?: number,
+  username?: string,
+  password?: string,
+  created_at?: number,
```

Comment:
So a provider record returned from Postgres should always have this value since it gets a default value - so is this the type for data about to be inserted to Postgres, where this value and others could be undefined?

Reply:
I need to look at this a bit more closely - the issue is that we need a record type for 'insertion records' that haven't had the defaults applied - it's entirely valid to have a record object without created_at as we can rely on default table behavior.  

 This needs to be moved to the db package in either case.

Reply:
I'd kind of like to name this `PostgresProviderInsertData` or something to make it clear what it is and why it is different than records returned from Postgres

Reply:
I believe this comment is outdated based on our hangout/discussion c591c12ea